pkg/apimachinery/discoverysummarizer/discoverysummarizer.go

```diff
@@ -0,0 +1,166 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package discoverysummarizer
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"k8s.io/kubernetes/pkg/api/unversioned"
+)
+
+// The list of servers as read from the config file.
+var serverList FederatedServerList
+
+var groupVersionPaths = map[string][]string{}
+var legacyVersionPaths = map[string][]string{}
+
+// Brings up the server at the given port, which provides a summarized view of
+// all group versions supported by the servers mentioned in the given config file.
```

Comment:
I think this does too much-- it should be factored roughly so that the main() looks like below:

```
cfg := readConfigFile()
server := discover.NewServer(cfg)
server.Run(port)
```